client/packages/system/src/Patient/ListView/ListView.tsx

```diff
@@ -70,7 +70,7 @@ const PatientListComponent: FC = () => {
         isError={isError}
         onRowClick={row => {
           if (!row.id || !row.document?.name || !row.document?.type) alert();
-          else navigate(`/patients/${row.id}`);
+          else navigate(`/patients/detail?doc=${row.document.name}`);
```

Comment:
ok, so it's nit-picky, but the url makes me sad.

I don't like having the implementation specific detail of `doc=` in there 🤷 but I understand that there's a need for it to be in there

As far as the order & contents: I would  prefer `/patients/${row.id}?doc=${document.type}`
We have 
- `patients` explaining the section
- [patient id] to get the detail
- some specifics in the query

this is the format elsewhere, for example `/distribution/outbound-shipment/2?sort=itemName`

Reply:
Yes, I agree that is nicer. The only reason I haven't done it like that so far is that it would require two database queries, which I've been trying to avoid initially.

We'd need to look up the document name from the patient record, then pass that name into JsonForms to fetch the document. Yes, we can get document data directly by drilling down into Patient node, but the JsonForms module is supposed to be agnostic to what kind of document it's looking for, so shouldn't know to query the Patient node.

Here's a couple of possible options I can think of to deal with it, you may have some other suggestions:

1. We don't do database lookups in `useJsonForms`. The container element would be responsible for fetching the document and schemas, then just pass them into `useJsonForms` for display. It makes saving more inconsistent though, as the "save" method should probably be in `useJsonForms` as it's super-generic and applies to documents of any type.
2. Just do two queries as mentioned above.
3.  Do a single database lookup in `useJsonForms`, but the container component passes in the database query method (and have a default query where it gets it from "document" node if none provided, which would be the current query)

I can see pros and cons to each, though option 2 is probably the most straightforward. Maybe chat about it on Monday?